presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SingleMarkDistinctToGroupBy.java

```diff
@@ -82,6 +82,9 @@ public Pattern getPattern()
             return Optional.empty();
         }
 
+        if (!child.getOrderBySymbols().isEmpty()) {
```

Comment:
The thing which bothers me is the fact that `MarkDistinctNode` contains `orderBySymbols`. Each `planNode` represents some SQL relation algebra function. To me `orderBySymbols` does not match the semantic of `MarkDistinctNode` unless the semantic has changed. 

To me `MarkDistinctNode` sets `markerSymbol` to true if values for `distinctSymbols` are not yet seen. How `orderBySymbols` could be used with that?

This is the only usage of `orderBySymbols` I found, I prefer to use `orderBySymbols` from parent and remove them from `MarkDistinctNode`.